tooling/cli/src/mobile/android/project.rs

```diff
@@ -122,14 +122,21 @@ pub fn gen(
       }
 
       let mut options = fs::OpenOptions::new();
+      options.write(true);
 
       #[cfg(unix)]
       if path.file_name().unwrap() == OsStr::new("gradlew") {
         use std::os::unix::fs::OpenOptionsExt;
         options.mode(0o755);
       }
 
-      options.create_new(true).write(true).open(path)
+      if path.file_name().unwrap() == OsStr::new("BuildTask.kt") {
+        options.create(true).open(path).map(Some)
+      } else if path.exists() {
+        Ok(None)
+      } else {
+        options.create(true).open(path).map(Some)
```

Comment:
this is the same as the block for the first `if`, can't you just change it to 
```rust
if path.file_name().unwrap() == OsStr::new("BuildTask.kt") || !path.exists() {
  options.create(true).open(path).map(Some)
} else {
  Ok(None)
}
```

Reply:
yeah that works